extensions-support/xalan/deployment/src/main/java/org/apache/camel/quarkus/support/xalan/deployment/XalanNativeImageProcessor.java

```diff
@@ -52,4 +55,29 @@ ReflectiveClassBuildItem reflectiveClasses() {
         return Arrays.asList(
                 new NativeImageResourceBuildItem("org/apache/xml/serializer/output_xml.properties"));
     }
+
+    @BuildStep
+    void installTransformerFactory(
+            CamelXalanBuildTimeConfig config,
+            BuildProducer<SystemPropertyBuildItem> properties,
+            BuildProducer<ServiceProviderBuildItem> serviceProviders,
+            BuildProducer<NativeImageResourceBuildItem> nativeResources) {
+
+        config.transformerFactory
+                .ifPresent(val -> properties.produce(
+                        /*
+                         * If we do not do this, the service provider defined in xalan.jar's
+                         * META-INF/services/javax.xml.transform.TransformerFactory
+                         * wins over our factory on Java 11+ native
+                         * I any case, the user has an option to pass his preferred factory instead of ours
+                         */
+                        new SystemPropertyBuildItem("javax.xml.transform.TransformerFactory", val)));
```

Comment:
It was an issue even before but this does not apply to JVM so we could have different behavior between native and jvm

Reply:
> this does not apply to JVM

It does. There is `NativeImageSystemPropertyBuildItem` that applies to native only.

Reply:
ah sorry I misread it, my bad